src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1567 @@
+            graphic_layers=graphic_layers,
+            graphic_groups=graphic_groups,
+            concept_name_code=concept_name_code,
+            institution_name=institution_name,
+            institutional_department_name=institutional_department_name,
+            content_creator_name=content_creator_name,
+            content_creator_identification=content_creator_identification,
+            rescale_intercept=rescale_intercept,
+            rescale_slope=rescale_slope,
+            rescale_type=rescale_type,
+            modality_lut=modality_lut,
+            copy_modality_lut=copy_modality_lut,
+            softcopy_voi_luts=softcopy_voi_luts,
+            copy_voi_lut=copy_voi_lut,
+            icc_profile=None,
+            transfer_syntax_uid=transfer_syntax_uid,
+            **kwargs
+        )
+
+        # Softcopy Presentation LUT
+        if presentation_luts is not None:
+            if presentation_lut_shape is not None:
+                raise TypeError(
+                    'Only one of "presentation_luts" or '
+                    '"presentation_lut_shape" should be provided.'
+                )
+            if len(presentation_luts) == 0:
+                raise ValueError(
+                    'Argument "presentation_luts" must not be empty.'
+                )
+            for v in presentation_luts:
+                if not isinstance(v, LUT):
+                    raise TypeError(
+                        'Items of "presentation_luts" should be of type '
+                        'highdicom.LUT.'
+                    )
+            self.PresentationLUTSequence = presentation_luts
+        else:
+            presentation_lut_shape = (
+                presentation_lut_shape or
+                PresentationLUTShapeValues.IDENTITY
+            )
+            self.PresentationLUTShape = PresentationLUTShapeValues(
+                presentation_lut_shape
+            ).value
+
+
+class PseudoColorSoftcopyPresentationState(_SoftcopyPresentationState):
+
+    """SOP class for a Pseudo-Color Softcopy Presentation State object.
+
+    A Pseudo-Color Softcopy Presentation State object includes instructions for
+    the presentation of a grayscale image as a color image by software.
+
+    """
+
+    def __init__(
+        self,
+        referenced_images: Sequence[Dataset],
+        series_instance_uid: str,
+        series_number: int,
+        sop_instance_uid: str,
+        instance_number: int,
+        manufacturer: str,
+        manufacturer_model_name: str,
+        software_versions: Union[str, Tuple[str]],
+        device_serial_number: str,
+        content_label: str,
+        red_palette_color_lut_data: np.ndarray,
+        green_palette_color_lut_data: np.ndarray,
+        blue_palette_color_lut_data: np.ndarray,
+        red_first_mapped_value: int,
+        green_first_mapped_value: int,
+        blue_first_mapped_value: int,
+        palette_color_lut_uid: Union[UID, str, None] = None,
```

Comment:
I definitely agree that this would make sense, but I'm not sure about the best way to do it. Most things that we have separated out into highdicom classes also belong in Sequences, meaning that we can just derive from Dataset or Sequence and place the resulting object into the top level Dataset being constructed.

However, the Paletter Color Lookup Table attributes all live at the root level of the dataset so we cannot take this approach. Should we just derive from Dataset anyway, and copy over the relevant attributes from it to the top level dataset, or write another class derived from plain object that basically "carries" the relevant information. 

What do you think?

Reply:
True, I forgot that they often are at the root level. In the VL Whole Slide Microscopy Image IOD, they are encapsulated in the Palette Color Lookup Table Sequence. We could implement this Sequence and then just copy the attributes into the target dataset under the hood. I think this may be the cleanest. What do you think?

We do take a similar approach with some of the SR content items.

Reply:
This is now implemented as suggested, and also added as an option to the Parametric Map class